clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml

```diff
@@ -2221,3 +2221,11 @@
       Out: ""
       MatchMode: Full
       RuleId: "remove_empty_library_arg"
+
+- Rule: rule_set_source_files_properties_append
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: set_source_files_properties_lang_append
+  In: set_source_files_properties(${src_files} PROPERTIES${some_other_prop}LANGUAGE CUDA${other_props})
```

Comment:
Match mode is too specific, cannot cover cases like:
set_source_files_properties(${src_files} PROPERTIES COMPILE_FLAGS "--some-flag" LANGUAGE C CUDA COMPILE_DEFINITIONS).

Reply:
That is a good point. It didn't occur to during my first thought.

But when I think about it can a same be marked with multiple languages?

Reply:
I just verifed that CMake complaints for such invocation of `set_source_files_properties`. see below.

```bash
shliclel672:/tmp/dhr/build
$ cat ../CMakeLists.txt
cmake_minimum_required(VERSION 3.3)
project(MulLang)

set(SOURCE_FILE "src.xx")
add_executable(MulLang ${SOURCE_FILE})

set_source_files_properties(${SOURCE_FILE} PROPERTIES LANGUAGE CXX C)

shliclel672:/tmp/dhr/build
$ cmake ..
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:7 (set_source_files_properties):
  set_source_files_properties called with incorrect number of arguments.


-- Configuring incomplete, errors occurred!
See also "/tmp/dhr/build/CMakeFiles/CMakeOutput.log".

shliclel672:/tmp/dhr/build
$
```